app/services/katello/pulp3/repository_mirror.rb

```diff
@@ -110,8 +110,10 @@ def fetch_remote
         api.remotes_list(name: backend_object_name).first
       end
 
-      def sync
-        repository_sync_url_data = api.class.repository_sync_url_class.new(remote: remote_href, mirror: true)
+      def sync(options = {})
+        options.delete(:optimize) unless repo.content_type == "yum"
+        options.delete(:skip_types) unless repo.content_type == "yum"
```

Comment:
could we move the knowledge about these options into the pulp3 yum repo service?

Also i'm surprised 'skip_types' was being passed in, that shouldn't ever be used with a smart proxy sync

Reply:
Sure thing.  I was deleting skip_types just because I saw it was a possible parameter for yum repos.  I can remove that deletion if skip_types will never be there.

Reply:
its fine to delete, i was more curious if you actually saw it being passed in, as that could indicate something unexpected was going on

Reply:
Think I've fixed it.  Not able to test functionally yet due to my broken proxy env, but the tests pass at least.

Reply:
Tested the functionality, still works